### ISSUE TITLE
Adding GitHub Actions

### ISSUE THREAD

**Issue by williamfgc** [QMCPACK/qmcpack#3020]
As brought up by @prckent . This is issue is regarding on exploring the addition of GitHub Actions to QMCPACK to current CI checks. The goal is to open discussion on the community on the scope of current and future quality checks (e.g. sanitizers, code coverage, etc.) as we go on.

**Comment by williamfgc** [comment]
The docker container with dependencies is located in my local account [here](https://hub.docker.com/repository/docker/williamfgc/qmcpack-ci). After discussion, we can move it to a more project account on DockerHub.

**Comment by prckent** [comment]
I think the key thing is to get the initial action up and running so that we can study the resource usage, learn more about github actions, and gradually expand add capabilities etc.

**Comment by williamfgc** [comment]
@prckent that's correct, thanks!  The first workflow is ready to go we just need to allow external actions (as in one-line workflows like checking out code, perform clang-format checks, etc.), see [here](https://github.com/QMCPACK/qmcpack/actions/runs/666169074). In order to get external actions (either official actions or from marketplace) working we can do `Settings>Actions` and either: 
1. `Allow all actions`
2.`Allow selected actions` and add `actions/checkout@v1`.  

Essentially, we either allow all actions or cherry-pick the ones we need as we go. I usually go with the former to set and forget, thoughts?

**Comment by williamfgc** [comment]
The first GitHub Actions workflow is up, see [here](https://github.com/QMCPACK/qmcpack/runs/2149217063?check_suite_focus=true). QMCPACK takes 25 minutes to build and deterministic tests took 3.5 min, here is the [Linux hardware](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources). @prckent we can follow up on the next steps (more compiler combinations, tests, sanitizers, nightly, etc.), thanks.

**Comment by prckent** [comment]
These timings seem about right for a build on one or few threads. Can the build be sped up? On a strong workstation  (make -j 32, flash filesystem etc.) the build time is a 2-3 minutes. Perhaps we only get one or a few threads, but ctest -j options might be adjustable. Our tests are generally built for up to 16 threads / cores total.

**Comment by williamfgc** [comment]
I think we can close this issue.
